### PR TITLE
[BAC-160] - fix elasticsearch k8s service

### PR DESCRIPTION
**Pull request by alika**
**What this PR does / why we need it**: This PR changes the k8s elasticsearch service into a [headless service][1]. This will allow the dns name of `elasticsearch` to properly resolve to one of the elasticsearch data nodes.

This PR also adds a helm test to verify the installed chart by `curl`ing the [elasticsearch cluster API][2]. In addition to adding this test to the gitlab CI/CD pipeline. In regards to the gitlab pipeline I also provisioned the `TEST_KUBECONFIG` (_onondaga_) secret variable for this project.

**Which issue this PR fixes**:
This is part of [BAC-160][160]

**Verification Notes**:

To verify you can use `build/test-install.sh` to install, test, and cleanup this chart into a dev cluster.

[1]: https://kubernetes.io/docs/concepts/services-networking/service/#headless-services
[2]: https://www.elastic.co/guide/en/elasticsearch/reference/5.2/cluster-health.html
[160]: https://samsung-cnct.atlassian.net/browse/BAC-160

### CODE REVIEW

File: elasticsearch-chart/templates/es-master-statefulset.yaml [hunk 1]
```diff
@@ -2,7 +2,6 @@ apiVersion: apps/v1beta1
 kind: StatefulSet
 metadata:
   name: {{.Values.master_name}}
-  namespace: {{.Values.namespace}}
```
Comment (coffeepac) [inline elasticsearch-chart/templates/es-master-statefulset.yaml]: why are these coming out?  I think we want to be able to specify what namespace to install into.

Comment (alika) [reply elasticsearch-chart/templates/es-master-statefulset.yaml]: Helm will do automatically do this for us if you use the option on the install command (`helm install --namespace ${NAMESPACE}`).  in that way we do not have to explicitly template that variable everywhere. In addition we can refer to `{{.Release.Namespace}}` to refer to the namespace helm received in our templates when needed explicitly, like our ClusterRoleBinding.

Comment (coffeepac) [reply elasticsearch-chart/templates/es-master-statefulset.yaml]: baller.

**Comment by alika** [comment]
@coffeepac I think this is ready for a final review and merge.

**Pull request merged**